### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by richbayliss**
Started to add typescript support and convert some of the files.

### CODE REVIEW

File: bin/katapult [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env node
 
-require('../src/lib/cli/katapult')
+require('../out/lib/cli/katapult')
```
[the remark this answers is not in the document]
Comment (richbayliss) [reply bin/katapult]: good to know, will make the change 👌

File: package.json [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
+		"mustache": "~2",
+		"mv": "~2",
+		"node-forge": "^0.7.6",
+		"simple-git": "~1",
+		"yamljs": "~0.3"
+	},
+	"devDependencies": {
+		"@types/bluebird": "^3.5.24",
+		"@types/lodash": "^4.14.118",
+		"@types/node": "^10.12.10",
```
[the remark this answers is not in the document]
Comment (richbayliss) [reply package.json]: Done

File: src/lib/controllers/deploy-adapters/index.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		super(`Unknown target: ${target}`);
+	}
+}
+
+export function getDeployAdapter(
+	target: string,
+	config: DeployConfig,
+): Deployable {
+	switch (target.toLowerCase().trim()) {
+		case 'balena':
```
Comment (mikesimos) [inline src/lib/controllers/deploy-adapters/index.ts]: We should probably need to handle `balena` via the balena deploy adapter here or remove it. (I see this was probably lost in the review process).

Comment (richbayliss) [reply src/lib/controllers/deploy-adapters/index.ts]: np, I will drop it down to the unsupported option for now.

**Pull request closed without merging**